Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,554 @@
+    intro o ho hgxo
+    rw [tsupport, _root_.mem_closure_iff] at hx
+    obtain ⟨y, hy⟩ := hx (g ⁻¹' o) (IsOpen.preimage g.1.2 ho) hgxo
+    exact ⟨g y, hy⟩
+
+@[simp]
+theorem coe_comp_to_continuous_fun (f : C_c(γ, δ)) (g : β →co γ) : ((f.comp g) : β → δ) = f ∘ g :=
+  rfl
+
+@[simp]
+theorem comp_id (f : C_c(γ, δ)) : f.comp (CocompactMap.id γ) = f :=
+  ext fun _ => rfl
+
+variable [T2Space β]
+
+@[simp]
+theorem comp_assoc (f : C_c(γ, δ)) (g : β →co γ) (h : α →co β) :
+    (f.comp g).comp h = f.comp (g.comp h) :=
+  rfl
+
+@[simp]
+theorem zero_comp (g : β →co γ) : (0 : C_c(γ, δ)).comp g = 0 :=
+  rfl
+
+end
+
+variable [T2Space γ]
+
+/-- Composition as an additive monoid homomorphism. -/
+def compAddMonoidHom [AddMonoid δ] [ContinuousAdd δ] (g : β →co γ) : C_c(γ, δ) →+ C_c(β, δ) where
+  toFun f := f.comp g
+  map_zero' := zero_comp g
+  map_add' _ _ := rfl
+
+/-- Composition as a semigroup homomorphism. -/
+def compMulHom [MulZeroClass δ] [ContinuousMul δ] (g : β →co γ) : C_c(γ, δ) →ₙ* C_c(β, δ) where
+  toFun f := f.comp g
+  map_mul' _ _ := rfl
+
+/-- Composition as a linear map. -/
+def compLinearMap [AddCommMonoid δ] [ContinuousAdd δ] {R : Type*} [Semiring R] [Module R δ]
+    [ContinuousConstSMul R δ] (g : β →co γ) : C_c(γ, δ) →ₗ[R] C_c(β, δ) where
+  toFun f := f.comp g
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+
+/-- Composition as a non-unital algebra homomorphism. -/
+def compNonUnitalAlgHom {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring δ]
+    [TopologicalSemiring δ] [Module R δ] [ContinuousConstSMul R δ] (g : β →co γ) :
+    C_c(γ, δ) →ₙₐ[R] C_c(β, δ) where
+  toFun f := f.comp g
+  map_smul' _ _ := rfl
+  map_zero' := rfl
+  map_add' _ _ := rfl
+  map_mul' _ _ := rfl
+
+end CompactlySupportedContinuousMap
+
+namespace CompactlySupportedContinuousMapClass
+
+section Basic
+
+variable [Zero β] [TopologicalSpace β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+instance : CoeTC F (CompactlySupportedContinuousMap α β) :=
+  ⟨fun f =>
+    { toFun := f
+      continuous_toFun := map_continuous f
+      hasCompactSupport' := hasCompactSupport f }⟩
+
+/-- the product of `f : F` assuming `ContinuousMapClass F α γ` and `ContinuousSMul γ β` and
+`g : C_c(α, β)` is in `C_c(α, β)` -/
+instance [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] : SMul F C_c(α, β) where
+  smul f g :=
+    ⟨⟨fun x ↦ f x • g x, (map_continuous f).smul g.continuous⟩, g.hasCompactSupport'.smul_left⟩
+
+@[simp]
+theorem coe_smulc [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] (f : F) (g : C_c(α, β)) :
+    ⇑(f • g) = fun x => f x • g x :=
+  rfl
+
+@[simp]
+theorem smulc_apply [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] (f : F) (g : C_c(α, β)) (x : α):
+    (f • g) x = f x • g x :=
+  rfl
```

Comment:
These ones should be in the `CompactlySupportedContinuousMap` section, because the main player here is indeed `C_c(α, β)`.

Reply:
I move them back.